### PR TITLE
Add tag support to ec2_placement_group

### PR DESCRIPTION
**Pull request by harisokanovic**
##### SUMMARY
Allows users to optionally create EC2 placement groups with tags. Extend integration suite with additional test cases.

##### ISSUE TYPE
- Feature Pull Request

### CODE REVIEW

File: plugins/modules/ec2_placement_group.py [hunk 1]
```diff
@@ -43,6 +43,10 @@
     default: cluster
     choices: [ 'cluster', 'spread', 'partition' ]
     type: str
+  tags:
+    description:
+      - A dict of key value pairs to associate with the placement group
+    type: dict
```
Comment (markuman) [inline plugins/modules/ec2_placement_group.py:49]:
```suggestion
  tags:
    description:
      - A dict of key value pairs to associate with the placement group
    type: dict
    version_added: 8.1.0
```

File: plugins/modules/ec2_placement_group.py [hunk 2]
```diff
@@ -95,6 +99,13 @@
       description: PG strategy
       type: str
       sample: "cluster"
+    tags:
+      description: Tags associated with the placement group
+      type: dict
+      sample:
+        tags:
+          some: value1
+          other: value2
```
Comment (markuman) [inline plugins/modules/ec2_placement_group.py:108]:
```suggestion
    tags:
      description: Tags associated with the placement group
      type: dict
      sample:
        tags:
          version_added: 8.1.0
          some: value1
          other: value2
```

**Comment by markuman** [comment]
@harisokanovic Thanks for your contribution. We'll also need a changelog fragment. Otherwise, LGTM.

**Comment by harisokanovic** [comment]
> @harisokanovic Thanks for your contribution. We'll also need a changelog fragment. Otherwise, LGTM.

Hi Markus, thanks for looking it over. I pushed an updated branch with your suggestions and a changelog fragment.

File: changelogs/fragments/20240613_ec2_placement_group_tags.yml [hunk 3]
```diff
@@ -0,0 +1,2 @@
+minor_changes:
+  - ec2_placement_group - Added support for creating with ``tags``
```
Comment (markuman) [inline changelogs/fragments/20240613_ec2_placement_group_tags.yml:2]:
```suggestion
  - ec2_placement_group - Added support for creating with ``tags`` (https://github.com/ansible-collections/community.aws/pull/2081).
```

**Review by tremble — DISMISSED** [review]
Generally looks good.  Our's linter's complaining about some whitespace, but nothing major.

File: plugins/modules/ec2_placement_group_info.py [hunk 4]
```diff
@@ -102,6 +103,7 @@ def get_placement_groups_details(connection, module):
                 "name": placement_group["GroupName"],
                 "state": placement_group["State"],
                 "strategy": placement_group["Strategy"],
+                "tags": boto3_tag_list_to_ansible_dict(placement_group.get("Tags")),
```
[the remark this answers is not in the document]
Comment (harisokanovic) [reply plugins/modules/ec2_placement_group_info.py:115]: Hi Mark, fixed and pushed an updated branch. Thanks.

**Comment by harisokanovic** [comment]
Hi @tremble @markuman, I believe all feedback was addressed in the latest update, and the bots now appear happy. Please let me know if I missed anything.

File: plugins/modules/ec2_placement_group.py [hunk 5]
```diff
@@ -43,6 +43,11 @@
     default: cluster
     choices: [ 'cluster', 'spread', 'partition' ]
     type: str
+  tags:
+    description:
+      - A dict of key value pairs to associate with the placement group
+    type: dict
+    version_added: 9.0.0
```
Comment (markuman) [inline plugins/modules/ec2_placement_group.py:50]:
```suggestion
    version_added: 9.0.0
```

I think we can release it with 8.1.0

Comment (harisokanovic) [reply plugins/modules/ec2_placement_group.py]: Changed to version_added back to 8.1.0

**Review by alinabuzachis — APPROVED** [review]

**Comment by harisokanovic** [comment]
Two tests failed with this doc error:
```
ERROR: Found 1 validate-modules issue(s) which need to be resolved:
ERROR: plugins/inventory/aws_mq.py:0:0: invalid-documentation: DOCUMENTATION.options.region.env.1.deprecated.Any('alternatives', 'alternative', msg=None): required key not provided @ data['options']['region']['env'][1]['deprecated'][Any('alternatives', 'alternative', msg=None)]. Got None
```

I don't think it's related to my changes since I did not edit aws_mq.py. Any suggestions on how to clear this up?

**Review by markuman — APPROVED** [review]

**Pull request merged**